### DISCUSSION TITLE
Internationalization

### DISCUSSION THREAD

**Discussion by karlguillotte** [contember/admin#714] [Support] [answered]
Me again ! This time, with a question regarding internationalization of the admin interface as well as the content.

In the [documentation](https://docs.contember.com/#main-features), we can read; 

**Multi-language, translatable**	

> Build global-ready apps with ease thanks to multi-language support.

I did look into more details in the documentation and at the examples, I can not find anything about the kind of support the framework is providing. I guess we can roll our own, but base on the quote above, I had the feeling that some kind of support is available out of the box. Is it a false assumption ? If not, could point me to more details. 

Again, many thanks for your help !!

**Comment by jonasnobile** [comment, marked as answer]
If you'd like to reflect internationalisation in your API model, you can combine our Dimensions with the i18n React library to display the correct language for each language dimension. Dimensions are a general concept that can be used for any time of dimensions, such as branches, countries or any other groups.

- [API model example for dimensions](https://github.com/contember/interface/blob/main/packages/playground/api/model/Dimensions.ts)
- [Dimensions component](https://github.com/contember/interface/blob/main/packages/playground/admin/app/pages/dimensions.tsx)

If you're not keen on using dimensions in your API model, no worries! You can simply implement internationalisation with any React library (Contember Admin is almost pure React app).

All the strings used in our new headless components are [`/admin/lib/dict.ts`](https://github.com/contember/examples/blob/next/template/admin/lib/dict.ts).

The documentation could be clearer. Internationalisation is about the API and Dimensions. The introduction is from when we were trying to be the backend of web applications. We've got lots more for you to do in our admin app now, and it's all ready to use as a great user interface.

We'll get the documentation changed as soon as we can to make it less confusing.
